File: erpmax/public/js/feature_toggle.js
frappe.provide("erpmax.feature");

erpmax.feature.get_toggles = function(callback) {
    if (erpmax.feature._toggles) {
        if (callback) callback(erpmax.feature._toggles);
        return;
    }
    frappe.call({
        method: "erpmax.erpmax.utils.feature_toggle.get_feature_toggles",
        callback: function(r) {
            erpmax.feature._toggles = r.message || {};
            if (callback) callback(erpmax.feature._toggles);
        }
    });
};

erpmax.feature.is_enabled = function(feature_key, callback) {
    erpmax.feature.get_toggles(function(toggles) {
        if (callback) callback(!!toggles["fea_" + feature_key]);
    });
};

File: erpmax/utils/feature_toggle.py
import frappe

@frappe.whitelist(allow_guest=False)
def get_feature_toggles(company=None):
    settings = frappe.get_cached_doc("ERPMax Settings")
    toggles = {}
    for field in settings.meta.fields:
        if field.fieldname.startswith("fea_"):
            toggles[field.fieldname] = settings.get(field.fieldname, 0)
    toggles["enable_commercial_terms"] = settings.get("enable_commercial_terms", 0)
    return toggles

def is_enabled(feature_key, company=None):
    settings = frappe.get_cached_doc("ERPMax Settings")
    return bool(settings.get(f"fea_{feature_key}", 0))

def is_commercial_terms_enabled(company=None):
    settings = frappe.get_cached_doc("ERPMax Settings")
    return bool(settings.get("enable_commercial_terms", 0))
